classes/model/slideshow.model.php

```diff
@@ -134,6 +135,27 @@ protected function duplicateSlideImages(Model_Slideshow $slider, Model_Slideshow
              */
             $clonedImage = clone $image;
             $clonedImage->slidimg_slideshow_id = $duplicatedSlider->slideshow_id;
+
+            /*
+             * For any given image containing a link to another site's page
+             * we find out if a translated page already exists on the target context.
+             * If so, we update the id of the linked page before save.
+             */
+            if ($clonedImage->slidimg_link_to_page_id) {
```

Comment:
I just noticed that there is a `page` relation on `Model_Image`. So you can do it in 3 lines :
```php
if ($clonedImage->page) {
    $clonedImage->page = $clonedImage->page->find_context($duplicatedSlider->slideshow_context) ?: $clonedImage->page;
}
```

Code not tested, but it should work.